src/community/wps-download/src/main/java/org/geoserver/wps/gs/download/DownloadProcess.java

```diff
@@ -255,26 +257,27 @@ public File execute(
                 LOGGER.log(Level.FINE, "Preparing the result");
             }
             // build output zip
-            final File result = resourceManager.getOutputResource(
-                    resourceManager.getExecutionId(true), resourceInfo.getName() + ".zip").file();
+            final Resource result = resourceManager.getOutputResource(
+                    resourceManager.getExecutionId(true), resourceInfo.getName() + ".zip");
 
-            FileOutputStream os1 = null;
+            OutputStream os1 = null;
```

Comment:
Opportunity for try-with-resources, but this one is a bit more complicated and may require advice from a module maintainer or @aaime . Is overriding AutoCloseable.close better or worse? Out of scope for this PR.